node/consensus/src/lib.rs

```diff
@@ -146,13 +146,14 @@ impl<N: Network, C: ConsensusStorage<N>> Consensus<N, C> {
     /// Advances the ledger to the next block.
     pub fn advance_to_next_block(&self, block: &Block<N>) -> Result<()> {
         // Adds the next block to the ledger.
+        let old_epoch = self.ledger.latest_epoch_number();
         self.ledger.advance_to_next_block(block)?;
```

Comment:
When `advance_to_next_block` is Ok, `block.epoch_number()` will always equal to `self.ledger.latest_epoch_number()`. Thanks to `check_block_basic` before `advance_to_next_block`, this doesn't cause any security problem, otherwise, it will cause chain fork